### PR TITLE
Build System.Private.DataContractSerialization on uapaot

### PR DESCRIPTION
**Pull request by shmao**
The fix is to get System.Private.DataContractSerialization to have a uapaot build.

### CODE REVIEW

File: src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ExtensionDataObject.cs [hunk 1]
```diff
@@ -13,15 +13,15 @@ public sealed class ExtensionDataObject
     {
         private IList<ExtensionDataMember> _members;
 
-#if USE_REFEMIT
+#if USE_REFEMIT || NET_NATIVE
```
Comment (joperezr) [inline src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ExtensionDataObject.cs]: Do we even need the USE_REFEMIT #ifdefs? It doesn't look like we are defining them on the csproj at all so I wonder if we should just clean those out and keep only the NET_NATIVE ones.

Comment (shmao) [inline src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ExtensionDataObject.cs]: I'd prefer keep them as is for now as we have other directives like this in other files. We can clean them all later.

Comment (weshaggard) [inline src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ExtensionDataObject.cs]: In general we should also remove NET_NATIVE ifdefs and rename it to uapaot.

Comment (shmao) [inline src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ExtensionDataObject.cs]: @weshaggard I opened https://github.com/dotnet/corefx/issues/15892 for tracking both of these. Just to clarify, do we want to use "uapaot" or "UAPAOT"?

**Pull request merged**